modules/installation-osp-local-disk-deployment.adoc

```diff
@@ -0,0 +1,365 @@
++
+[%collapsible]
+====
+[source,terminal]
+----
+$ oc patch ControlPlaneMachineSet/cluster -n openshift-machine-api --type json -p ' <1>
+[
+    {
+      "op": "add",
+      "path": "/spec/template/machines_v1beta1_machine_openshift_io/spec/providerSpec/value/additionalBlockDevices", <2>
+      "value": [
+        {
+          "name": "etcd",
+          "sizeGiB": 10,
+          "storage": {
+            "type": "Local" <3>
+          }
+        }
+      ]
+    }
+  ]
+'
+----
+<1> Applies the JSON patch to the `ControlPlaneMachineSet` CR.
+<2> Specifies the path where the `additionalBlockDevices` are added.
+<3> Adds the `etcd` devices with local storage of 10 GB to the cluster.
+====
+
+. Verify that the control plane machines are healthy:
+
+.. Verify the cluster is progressing by running the following command:
+
++
+[source,terminal]
+----
+$ oc wait --timeout=90m --for=condition=Progressing=false controlplanemachineset.machine.openshift.io -n openshift-machine-api cluster
+----
+
+.. Verify that 3 replica control plane machine sets are running by running the following command:
+
++
+[source,terminal]
+----
+$ oc wait --timeout=90m --for=jsonpath='{.spec.replicas}'=3 controlplanemachineset.machine.openshift.io -n openshift-machine-api cluster
+----
+
+.. Verify that the 3 control plane machine sets are updated by running the following command:
+
++
+[source,terminal]
+----
+$ oc wait --timeout=90m --for=jsonpath='{.status.updatedReplicas}'=3 controlplanemachineset.machine.openshift.io -n openshift-machine-api cluster
+----
+
+.. Verify that the 3 control plane machine sets are healthy by running the following command:
+
++
+[source,terminal]
+----
+$ oc wait --timeout=90m --for=jsonpath='{.status.replicas}'=3 controlplanemachineset.machine.openshift.io -n openshift-machine-api cluster
+----
+
+.. Verify the `ClusterOperators` are progressing in the cluster by running the following command:
+
++
+[source,terminal]
+----
+$ oc wait clusteroperators --timeout=30m --all --for=condition=Progressing=false
+----
+
+. Verify that each of the 3 control plane machines has the additional block device you previously created by running the following script:
+
++
+[source,bash]
+----
+$ cp_machines=$(oc get machines -n openshift-machine-api --selector='machine.openshift.io/cluster-api-machine-role=master' --no-headers -o custom-columns=NAME:.metadata.name) <1>
+
+
+if [[ $(echo "${cp_machines}" | wc -l) -ne 3 ]]; then
+  exit 1
+fi <2>
+
+
+for machine in ${cp_machines}; do
+  if ! oc get machine -n openshift-machine-api "${machine}" -o jsonpath='{.spec.providerSpec.value.additionalBlockDevices}' | grep -q 'etcd'; then
+	exit 1
+  fi <3>
+done
+----
+<1> Retrieves the machines running in the cluster.
+<2> Checks that there are 3 machines running.
+<3> Checks that 3 `etcd` devices are running on the 3 machines.
+
+. Create a file named `98-var-lib-etcd.yaml` by using the following content:
+
+[WARNING]
```

Comment:
@mdbooth does this warning help?

Reply:
My worry is that users have been known to cargo-cult knowledge like this. If we give the impression that moving etcd is dangerous, even in TP, it might be hard to convince folks to do it in production later after we've fixed the bugs.

This warning suggests that any etcd move operation must be done on a non-production cluster. That isn't true in general, and it won't be true of the version we will have to write before GA. It is only true of *this* documented move operation because of the identified bugs.

Can we write the warning in such a way that it articulates clearly:
* The procedure documented here is useful for testing the performance of etcd on local disk on a non-production cluster
* The procedure documented here has known bugs which will eat your cluster in certain circumstances
* These bugs will not be present at GA, at which time this operation will be safe

Reply:
> Can we write the warning in such a way that it articulates clearly:
> 
> * The procedure documented here is useful for testing the performance of etcd on local disk on a non-production cluster
> * The procedure documented here has known bugs which will eat your cluster in certain circumstances
> * These bugs will not be present at GA, at which time this operation will be safe

@mdbooth I'm a little confused because you write that `This warning suggests that any etcd move operation must be done on a non-production cluster. That isn't true in genral...`; however, you also ask that the warning be changed to include `The procedure documented here is useful for testing the performance of etcd on local disk on a non-production cluster`.

With respect to your question about re-writing the warning. The short answer is we can do the first bullet point, the second and third we can not. We can't make promises about future performance or support. We could if you'd like write up bugs in the RNs or KIs if you'd like to warn our customers about the situation.

Reply:
@mdbooth docs freeze is 2/13 so we are on a very tight schedule at this point. This will have to go through peer-review again and we still need QE review.